pkg/cluster/deploystorage.go

```diff
@@ -120,15 +120,22 @@ func (m *manager) deployStorageTemplate(ctx context.Context) error {
 		m.clusterNSG(infraID, azureRegion),
 		m.clusterServicePrincipalRBAC(),
 		m.networkPrivateLinkService(azureRegion),
-		m.networkPublicIPAddress(azureRegion, infraID+"-pip-v4"),
 		m.networkInternalLoadBalancer(azureRegion),
-		m.networkPublicLoadBalancer(azureRegion),
 	}
 
-	if m.doc.OpenShiftCluster.Properties.IngressProfiles[0].Visibility == api.VisibilityPublic {
+	// Create a public load balancer routing if needed
```

Comment:
So the ordering on this is interesting.  Can you append to your comment and add something like:
`m.determineOutboundType` will ensure that the outbound type is set correctly so we do not have to evaluate whether the visibility is public or private in this if statement since it's already been confirmed by the function.`

It's very order-specific and it's confusing to review/read in isolation.  We have to look at the installation as a whole to ensure this is doing what we expect.